### PR TITLE
Singularity, Narsie and exit wormhole rewrite.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
**Testing:**
- [X] Basic singularity spawning, growing, consuming, wandering.
- [X] Singularity chasing beacons.
- [x] Supermatter singularity.
- [x] Nar'sie
- [x] Exit wormhole

## Description of changes
Breaks up and rewrites the aforementioned.

## Why and what will this PR improve
Better code, easier to move Narsie into a modpack with cult.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Still need to test this in a context of building an engine but have not got a map to do that with.

**Comment by MistakeNot4892** [comment]
CI hangup is IDB.

File: code/game/gamemodes/cult/ritual.dm [hunk 1]
```diff
@@ -48,7 +48,7 @@
 	if(locate(/obj/effect/rune) in T)
 		to_chat(src, "<span class='warning'>There's already a rune here.</span>") // Don't cross the runes
 		return
-	if(T.icon_state == "cult" || T.icon_state == "cult-narsie")
+	if(T.icon_state == "cult" || (locate(/obj/effect/narsie_footstep) in T))
```
Comment (comma) [inline code/game/gamemodes/cult/ritual.dm]: T.is_defiled() ?

File: code/game/gamemodes/cult/narsie.dm [hunk 2]
```diff
@@ -1,362 +1,181 @@
...
+						targets += food
+			// Go for ghosts if nothing else is available.
+			if(!length(targets))
+				for(var/mob/observer/G in global.dead_mob_list_)
+					targets += G
+		// Pick a random candidate.
+		if(length(targets))
+			target = pick(targets)
+		var/capname = uppertext(name)
+		to_chat(target, SPAN_NOTICE("<b>[capname] HAS LOST INTEREST IN YOU.</b>"))
```
Comment (comma) [inline code/game/gamemodes/cult/narsie.dm]: Is this supposed to give both 'lost interest' and 'hungers for your soul' messages to the same dude?

Comment (MistakeNot4892) [reply code/game/gamemodes/cult/narsie.dm]: Ordering issue, fixed now.

**Review by comma — APPROVED** [review]

**Review by out-of-phaze — APPROVED** [review]
Merging since it's a prereq for moving gamemodes into modpacks.

**Pull request merged**